drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java

```diff
@@ -241,26 +254,56 @@ protected void onStartup(final ObservablePath path,
                              final PlaceRequest placeRequest) {
         this.editorPath = path;
         this.editorPlaceRequest = placeRequest;
+        registerDock( new DockPlaceHolderPlace(AnalysisDockPlaceHolder.IDENTIFIER,
+                                               VERIFIER_DOCK),
+                     analysisReportScreen.asWidget());
     }
 
     protected void onFocus() {
 
-        if (!docks.isSetup()) {
-            docks.setup(perspectiveManager.getCurrentPerspective().getIdentifier(),
-                        new DefaultPlaceRequest("org.kie.guvnor.explorer"));
-        }
-        docks.show();
-
         modeller.getActiveDecisionTable().ifPresent(dt -> {
             decisionTableSelectedEvent.fire(new DecisionTableSelectedEvent(dt));
-            dt.initialiseAnalysis();
+            dt.initialiseAnalysis(analysisReportScreen);
         });
     }
 
-    private void hideDataModellerDocks(@Observes PlaceHiddenEvent event) {
+    public void onShowDiagramEditorDocks(@Observes PlaceGainFocusEvent event) {
+        if (verifyEventIdentifier(event)) {
+            if (!docks.isSetup()) {
+                docks.setup(perspectiveManager.getCurrentPerspective().getIdentifier(),
+                            new DefaultPlaceRequest("org.kie.guvnor.explorer"));
+            }
+            showDocks();
+        }
+    }
+
+    public void onHideDocks(@Observes PlaceHiddenEvent event) {
+        if (verifyEventIdentifier(event)) {
+            hideDocks();
+        }
+    }
+
+    public void showDocks() {
+        docks.show();
+        guidedDecisionTableDocksHandler.addDocks();
+    }
+
+    public void hideDocks() {
         docks.hide();
+        guidedDecisionTableDocksHandler.removeDocks();
+    }
+
+    public boolean verifyEventIdentifier(AbstractPlaceEvent event) {
+        boolean equals = Objects.equals(getEditorIdentifier(),
+                                        event.getPlace().getIdentifier());
+        boolean equals1 = Objects.equals(editorPlaceRequest,
+                                         event.getPlace());
+        return (equals &&
+                equals1);
```

Comment:
Toni, could we please choose better variable names here? `equals` and `equals1` do not say much for newcomer to the codebase.

Or we could avoid storing partial results by:
```java
if(!Objects.equals(editorPlaceRequest, event.getPlace())) {
   return false;
}

if (!Objects.equals(getEditorIdentifier(), event.getPlace().getIdentifier())) {
    return false;
}
return true;
```

Please 
let me know what do you think.

Reply:
Both work. Looks like I left this in when debugging. Fixing.